sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -286,7 +286,7 @@ trait PrunedScan {
  */
 @DeveloperApi
 trait PrunedFilteredScan {
-  def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row]
+  def buildScan(requiredColumns: Seq[String], filters: Seq[Filter], aggregate: Aggregate): RDD[Row]
```

Comment:
Sorry and I'm not familiar this interface issue.
The fix satisfies your comment?
https://github.com/apache/spark/pull/10631/files#diff-40c347747af9101e7e9fee52fc4120b8R290